### PR TITLE
[MM-38891] - CRT: threads list loading indicator and search guidance …

### PR DESCRIPTION
**Pull request by julmondragon**
#### Summary
This PR adds two enhancements to the CRT feature for the Threads global list:

- Adds a loading indicator each time more posts are being fetched on infinite scroll.
- When no more posts can be loaded (due to the historical limit of threads), provides search guidance at the bottom of the list. This is only shown if you’ve scrolled to load older posts and then reach the end of the list.

#### Ticket Link
Issue: https://github.com/mattermost/mattermost-server/issues/18796
JIRA: https://mattermost.atlassian.net/browse/MM-38891

#### Screenshots

MM-38891

#### Release Note
```release-note
CRT - Threads global list: Added a loading indicator each time more posts are being fetched on infinite scroll.
```

```release-note
CRT - Threads global list: Added a search guidance when no more posts can be loaded. This is only shown if you’ve scrolled to load older posts and then reach the end of the list.
```

### CODE REVIEW

**Comment by julmondragon** [comment]
FYI @neallred
@calebroseland could you please help reviewing?

File: components/threading/global_threads/thread_list/virtualized_thread_list.tsx [hunk 1]
```diff
@@ -46,7 +50,15 @@ function VirtualizedThreadList({
         }
     }, [selectedThreadId, ids]);
 
-    const data = useMemo(() => ({ids, selectedThreadId}), [ids, selectedThreadId]);
+    const data = useMemo(
+        () => (
+            {
+                ids: hasLoaded && ids.length === total ? [...ids, Constants.THREADS_NO_RESULTS_ITEM_ID] : (isLoading && ids.length !== total && [...ids, Constants.THREADS_LOADING_INDICATOR_ITEM_ID]) || ids,
```
Comment (neallred) [inline components/threading/global_threads/thread_list/virtualized_thread_list.tsx:56]: I saw an issue where the first attempt to load more wouldn't show the loading indicator, but subsequent loads would. I'm guessing its something related to the `isLoading` check here.

https://user-images.githubusercontent.com/13738432/146991231-26e44bac-7e9b-475f-9c08-c829990860a7.mp4

**Review by neallred — COMMENTED** [review]
It looks really nice! The only issues I see are:
1) With some screen widths, it would be nice to have the no results indicator be bigger to accommodate all the content
2) For the first time the virtualized list is loading, it does not show the loading dots.

I left more details as inline comments.

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
Thanks for working on this @julmondragon. Beyond what @neallred mentioned about the responsiveness, there are some other styling adjustments needed as outlined below:

1. Change the padding on the container to `padding: 16px 16px 16px 24px;`
2. Change font-size on "That's the end of the list" header to: `font-size: 16px;` and `line-height: 24px`
3. Make the margin on the bottom of the header `margin-bottom: 0px;`
4. Need to re-export the SVG to the clipped size provided in the [figma file at this link](https://www.figma.com/file/B1FyrTYjrfoA9AMCqIQ9tA/%E2%9C%85MM-38891-Threads-Infinite-Scroll-Loader?node-id=233%3A25099). Once you've done that, you should have an SVG that is 72x48. Let me know if you need me to export this for you.
5. Change margin-right on the SVG to `margin-right: 16px`
5. Do we need to set an explicit height on the container?  Or should it grow to the height it needs to automatically? It looks like the height is explicitly set to 133px. If this needs to be an explicit height, this height will likely need to change based on the feedback above.
6. The shortcut style is quite different from the design. Quite a few changes need to be made to those to align with design:
```
font-size:12px;
font-weight: 600;
line-height: 16px;
padding: 2px 4px;
color: rgba(var(--center-channel-color-rgb),.72);
```

**Comment by AshishDhama** [comment]
@julmondragon Thanks for working on this I think you are almost there, If you need some help on this one please let me know.

**Comment by esethna** [comment]
Hey @julmondragon, thanks for the PR! How's this going? Are you open to finishing this up?

**Comment by julmondragon** [comment]
Hi @esethna , yes! I'll resume the work on this ticket to address the feedback soon.

**Comment by esethna** [comment]
Thanks @julmondragon!

**Comment by julmondragon** [comment]
@neallred @matthewbirtch I think I addressed all the feedback received.

**Loading indicator and Search guidance**
CRT-loading-search guidance


**Screen sizes**
CRT-search guidance-dims

**Review by matthewbirtch — APPROVED** [review]
Thanks @julmondragon! Looks good to me

**Review by pablovelezvidal — APPROVED** [review]
LGTM! great work @julmondragon

**Review by neallred — APPROVED** [review]

**Review by stevemudie — APPROVED** [review]
Looks good @julmondragon!

Scrolling worked as expected and the information on how to search at the bottom of the list was there as well.

**Comment by esethna** [comment]
@julmondragon I think this is good to be merged, thanks for the work!

**Pull request merged**